pallets/rc-migrator/src/accounts.rs

```diff
@@ -519,25 +551,71 @@ impl<T: Config> AccountsMigrator<T> {
 	///
 	/// Should be executed once before the migration starts.
 	pub fn obtain_rc_accounts() -> Weight {
-		for (channel_id, _info) in hrmp::HrmpChannels::<T>::iter() {
+		let mut reserves = sp_std::collections::btree_map::BTreeMap::new();
+		let mut update_reserves = |id, deposit| {
+			if deposit == 0 {
+				return;
+			}
+			reserves.entry(id).and_modify(|e| *e += deposit).or_insert(deposit);
+		};
+
+		for (channel_id, info) in hrmp::HrmpChannels::<T>::iter() {
 			let sender: T::AccountId = channel_id.sender.into_account_truncating();
-			RcAccounts::<T>::insert(sender, AccountStateFor::<T>::Preserve);
+			update_reserves(sender, info.sender_deposit);
 
 			let recipient: T::AccountId = channel_id.recipient.into_account_truncating();
-			RcAccounts::<T>::insert(recipient, AccountStateFor::<T>::Preserve);
+			update_reserves(recipient, info.recipient_deposit);
+		}
+
+		for (channel_id, info) in hrmp::HrmpOpenChannelRequests::<T>::iter() {
+			let sender: T::AccountId = channel_id.sender.into_account_truncating();
```

Comment:
Can you add a link here please to reference where this account is calculated in the pallet?

Reply:
updated